### ISSUE TITLE
Design an enclosure

### ISSUE THREAD

**Issue by jcard0na** [cardonabits/haxo-hw#3]

**Comment by JasonShifflett** [comment]
This project looks awesome, would be interested in trading some design for some prototype access. :D

**Comment by jcard0na** [comment]
Hi, thanks for the encouragement.  I do not have a spare functioning prototype at the moment.  I do have one with a broken pressure sensor.  Would you be interested in that?  You could probably repair it yourself, and at any rate, the device does not need to be functional for you to design an enclosure... 😉

**Comment by JasonShifflett** [comment]
Well, that sounds like a cool project and a bundle of fun. Are there any initial design sketches? I’ve been futzing around with my own 3D printed designs, and I’d love to add one of these to my instruments.

**Comment by jcard0na** [comment]
Nope.  The original idea for the main PCB was to be the structural backbone of the instrument.  This has kept the cost down, leaves components accessible for repairs and highlights the openness of the project.  Designing an enclosure that respects those principles (low cost, repair-ability, openness), and considering that the initial design did not consider one, is definitely a challenge.  But hey, [constraints are supposed to be good for creativity](https://medium.com/upskilling/ideation-creativity-of-constraints-tells-us-that-limits-make-us-more-creative-5f6a82afaee4)!

**Comment by anoopnir** [comment]
Hi I Am interested in the project. I would like to contribute to the project by designing a beautiful case for the same. Would you like it in 3d or in physical form. Can it be 3dprinted or   what type you prefer . It's a marvelous peace of instrument as far as I have seen . Thanks you for the great effort for developing one like it.

**Comment by jcard0na** [comment]
Hi @anoopnir 

Thanks for the kind words.  As long as it is low cost, easy to repair, open and beautiful, you can take all the creative freedoms that you want.  3D printable or flat laser cut are the two technologies that I know would make it easy for people to build their own.

**Comment by anoopnir** [comment]
Dear friend I can create a case in 3d max only . Is it ok ? I don't possess solidworks  which is good in this case.  Please let me know.

**Comment by Utopiah** [comment]
Well I didn't go very far for now but here is a start : 

<img width="429" alt="image" src="https://github.com/cardonabits/haxo-hw/assets/1076655/84e6567a-54bb-44cb-9bca-3244fe7989e7">

Ok, there is nothing, just a scaffolding, but still it's an excuse to restart this thread.
[haxo.zip](https://github.com/cardonabits/haxo-hw/files/13900967/haxo.zip)

I imported as SVG the cut layer from KiCAD but this could instead rely on the KiCAD project via e.g https://github.com/30350n/pcb2blender , my recommendation : 

1. clone this repository
2. open the project in KiCAD
3. export the right layer in SVG
4. clean the layer in Inkscape
5. import in Blender

The STL imported come from https://github.com/cardonabits/haxo-hw/discussions/20 and there are more, e.g STEP files in https://github.com/cardonabits/haxo-hw/tree/main/models

Those STL should give the right size so shouldn't be resized.

The Pi Zero casing comes from https://www.thingiverse.com/thing:1165227

SVGs : 


![haxophone001-Edge_Cuts](https://github.com/cardonabits/haxo-hw/assets/1076655/fb30806e-2a5f-4781-b69d-68031248801d)
![haxo_bottom-side](https://github.com/cardonabits/haxo-hw/assets/1076655/164c6eed-53a7-4e72-b866-de8925a42f34)
![haxo_top-side](https://github.com/cardonabits/haxo-hw/assets/1076655/3c412030-5189-4a5d-8bb5-dfc97abe9472)

**Comment by Utopiah** [comment]
I have a 3D printer available so I'll try some casings. I have some transparent filament so might make for a good fit, to keep in mind that it's something that should still be tinkered with.

**Comment by Utopiah** [comment]
Apparently KiCAD 8 (rc1) https://gitlab.com/kicad/code/kicad/-/merge_requests/1253#note_1726636488 has glTF export. Maybe someone could produce a .glb to import from Blender, it would be a lot more precise.

**Comment by jcard0na** [comment]
Just to experiment, I installed kicad v8rc1 and generated this
```
kicad-cli pcb export glb --output haxophone.glb --subst-models --include-tracks --include-zones --grid-origin ~/dev/haxo-hw/haxophone001.kicad_pcb
```
Output attached.
[haxophone.glb.zip](https://github.com/cardonabits/haxo-hw/files/13952956/haxophone.glb.zip)

I confirmed that it can be imported into blender, and saw that only a few components are rendered.  If anyone plans to do something useful with this, I could try to spend more time and generate a more complete model.

**Comment by Utopiah** [comment]
FWIW AR preview https://twitter.com/utopiah/status/1753324762753802334 but still "just" of the export, not of the possible enclosure.

**Comment by Domi1l** [comment]
![WhatsApp Image 2024-06-14 at 16 08 42](https://github.com/cardonabits/haxo-hw/assets/164619675/0f152a55-7845-465e-ab3a-70075aa1e5cf)

So im on the way to finalize an enclosure. Hopefully its getting well.  :-)

**Comment by jcard0na** [comment]
Great progress!  It looks pretty funky, in a good way...  Is it possible/easy to round those edges?
![Screenshot 2024-06-15 at 12 16 38](https://github.com/cardonabits/haxo-hw/assets/676181/bca6eead-af1f-4183-adec-8a2560b6579e)

**Comment by Domi1l** [comment]
> Great progress! It looks pretty funky, in a good way... Is it possible/easy to round those edges? ![Screenshot 2024-06-15 at 12 16 38](https://private-user-images.githubusercontent.com/676181/339977018-bca6eead-af1f-4183-adec-8a2560b6579e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0NzQzNDUsIm5iZiI6MTcxODQ3NDA0NSwicGF0aCI6Ii82NzYxODEvMzM5OTc3MDE4LWJjYTZlZWFkLWFmMWYtNDE4My1hZGVjLThhMjU2MGI2NTc5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQxNzU0MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMjA2MWM5NWIyMTI5NWM0NjkzNzFjNGEyNTE3YTc4YjY4YmU0ODIxYmU1NzY3YTcxNTM0M2I5NTZkNzA5OTMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ExByNwl3Xp5V-hoXRcvbmJ3UYh9LdphWQdLXNZsq15Y)

Yes, this is just the "Prototype" to see if all openings and the dimensons are correct. Everything else like round edges and other ergonomic features are coming when the basic design is ready. If you have further suggestions you can tell me i try to build it in :-)

**Comment by jcard0na** [comment]
Excellent, thanks!

**Comment by Domi1l** [comment]
So, right now im printing the case in its final form (hopefully). I think now everything will work out to look fine and the positons are good. Dont mind my bad printing surfaces :-P 

![grafik](https://github.com/user-attachments/assets/2672f10d-0ac8-4025-922c-2ef296491bf0)

![grafik](https://github.com/user-attachments/assets/a54b70f2-3845-4893-a333-df2b28a0de13)

![grafik](https://github.com/user-attachments/assets/7c6b207a-5371-47c3-8fd1-131e361824a4)

If the printer is ready and everything works out i will post the STL.

Cheers!

**Comment by jcard0na** [comment]
Oh, funky!  Thanks for sharing!

**Comment by Domi1l** [comment]
So better late than never, sorry for the long waiting time. 

[https://www.thingiverse.com/thing:6737391](url)

Have a try :-)

**Comment by tuxiqae** [comment]
So, it's almost been a year since this design was released, how's it fairing so far? Should I try it?

**Comment by jcard0na** [comment]
> So, it's almost been a year since this design was released, how's it fairing so far? Should I try it?

I do not have a 3D printer and I never tried this.  But would love to hear from anyone using it.  So to answer your second question... If it is easy for you to print it, you should try it! 😉

**Comment by Utopiah** [comment]
I'm away from home but I could try 3D print in September and let you know.
